pygmo/common_utils.cpp

```diff
@@ -178,41 +178,78 @@ py::tuple inds_to_tuple(const pagmo::individuals_group_t &inds)
     return py::make_tuple(std::move(ID_arr), std::move(dv_arr), std::move(fv_arr));
 }
 
+// Convert a Python iterable into problem bounds.
+std::pair<pagmo::vector_double, pagmo::vector_double> iterable_to_bounds(const py::iterable &o)
+{
+    auto begin = std::begin(o);
+    const auto end = std::end(o);
+
+    if (begin == end) {
+        // Empty iterable.
+        py_throw(PyExc_ValueError, "cannot convert an empty iterable into problem bounds");
+    }
+
+    // Try fetching the lower bound
+    auto lb = ndarr_to_vector<std::vector<double>>(py::cast<py::array_t<double>>(*begin));
+
+    if (++begin == end) {
+        // iterable with only 1 element.
+        py_throw(PyExc_ValueError,
+                 "cannot convert an iterable with only 1 element into problem bounds (exactly 2 elements are needed)");
+    }
+
+    // Try fetching the upper bound
+    auto ub = ndarr_to_vector<std::vector<double>>(py::cast<py::array_t<double>>(*begin));
+
+    if (++begin != end) {
+        // iterable with too many elements.
+        py_throw(PyExc_ValueError, "cannot convert an iterable with more than 2 elements into problem bounds "
+                                   "(exactly 2 elements are needed)");
+    }
+
+    if (ub.size() != lb.size()) {
```

Comment:
Is it really necessary to have this check here? Aren't the genetic operators going to check this themselves?

Reply:
But this is in common utile, someone in the future may use it for other purposes?